pkg/arvo/sys/vane/gall.hoon

```diff
@@ -1575,52 +1549,18 @@
   ~|  [%gall-call-failed duct q.hic]
   =/  =task:able  ((harden task:able) q.hic)
   ::
-  =/  initialised  (mo-abed:mo duct)
+  =/  mo-core  (mo-abed:mo duct)
   ?-    -.task
-      %conf
-    =/  =dock  p.task
-    =/  =ship  p.dock
-    ?.  =(our ship)
-      ~&  [%gall-not-ours ship]
-      [~ gall-payload]
-    ::
-    =>  (mo-boot:initialised q.dock q.task)
-    mo-abet
-  ::
+      %conf  mo-abet:(mo-boot:mo-core dap.task our %home)
       %deal
-    =/  =sock  p.task
-    =/  =term  q.task
-    =/  =deal  r.task
+    =/  [=sock =term =deal]  [p q r]:task
```

Comment:
this is neat -- Hoon-style "destructuring!"

Reply:
Yeah, you can even do more complex tree shapes, as long as they're finite:
```hoon
=/  [[=ship =desk =case:clay] =spur]  *beam
`path`[(scot %p ship) desk (scot case) (spud (flop spur))]
```